Builds/Test.py

```diff
@@ -227,45 +327,168 @@ def run_build(args=None):
                 print(*lines, sep='')
             sys.exit(1)
 
+def get_cmake_dir(cmake_dir):
+    return os.path.join('build' , 'cmake' , cmake_dir)
+
+def run_cmake(directory, cmake_dir, args):
+    print('Generating build in', directory, 'with', *args or ('default options',))
+    old_dir = os.getcwd()
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+    os.chdir(directory)
+    if IS_WINDOWS and not any(arg.startswith("-G") for arg in args) and not os.path.exists("CMakeCache.txt"):
+        if '--ninja' in args:
+            args += ( '-GNinja', )
+        else:
+            args += ( '-GVisual Studio 14 2015 Win64', )
+    args += ( '-Dtarget=' + cmake_dir, os.path.join('..', '..', '..'), )
+    resultcode, lines = shell('cmake', args)
 
-def main():
-    if ARGS.all:
-        to_build = ALL_BUILDS
-    else:
-        to_build = [tuple(ARGS.scons_args)]
+    if resultcode:
+        print('Generating FAILED:')
+        if not ARGS.verbose:
+            print(*lines, sep='')
+        sys.exit(1)
+
+    os.chdir(old_dir)
+
+def run_cmake_build(directory, target, config, clean, args):
+    print('Building', target, config, 'in', directory, 'with', *args or ('default options',))
+    build_args=('--build', directory)
+    if target:
+      build_args += ('--target', target)
+    if config:
+      build_args += ('--config', config)
+    if clean:
+      build_args += ('--clean-first')
+    if args:
+        build_args += ('--',)
+        build_args += tuple(args)
+    resultcode, lines = shell('cmake', build_args)
```

Comment:
I'm not a cmake expert, but I thought one needs to run `cmake [args]` first, then run `make` or `ninja` because cmake just generates a `Makefile`? If one just needs to run cmake twice, this is of course correct.